strum_macros/src/lib.rs

```diff
@@ -409,6 +409,45 @@ pub fn enum_is(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
     toks.into()
 }
 
+/// Creates a new type that maps all the variants of an enum to another generic value.
+///
+/// This macro does not support any additional data on your variants.
+/// The macro creates a new type called `YourEnumTable<T>`.
+/// The table has a field of type `T` for each variant of `YourEnum`. The table automatically implements `Index<T>` and `IndexMut<T>`.
+/// ```
+/// use strum_macros::EnumTable;
+///
+/// #[derive(EnumTable)]
+/// enum Color {
+///     Red,
+///     Yellow,
+///     Green,
+///     Blue,
+/// }
+///
+/// assert_eq!(ColorTable::default(), ColorTable::new(0, 0, 0, 0));
+/// assert_eq!(ColorTable::filled(2), ColorTable::new(2, 2, 2, 2));
+/// assert_eq!(ColorTable::from_closure(|_| 3), ColorTable::new(3, 3, 3, 3));
+/// assert_eq!(ColorTable::default().transform(|_, val| val + 2), ColorTable::new(2, 2, 2, 2));
+///
+/// let mut complex_map = ColorTable::from_closure(|color| match color {
+///     Color::Red => 0,
+///     _ => 3
+/// });
+/// complex_map[Color::Green] = complex_map[Color::Red];
+/// assert_eq!(complex_map, ColorTable::new(0, 3, 0, 3));
+///
+/// ```
+#[proc_macro_derive(EnumTable, attributes(strum))]
```

Comment:
I don't feel too strongly, but rust tends to use the "Map" terminology rather than "Table." Thoughts about calling this `EnumMap` instead?

*I thoughts that's what the feature would be called based on the PR title.

Reply:
We had a conversation about this a while back and the conclusion was that `Map`s tend to be dynamically sized based on how full they are, whereas this type always has a value for each key. We thought these differences were enough that calling the structure a `Map` would cause people to make incorrect assumptions about how it works. That said, I don't feel too strongly about it either.